src/main/java/uk/gov/justice/hmpps/casenotes/services/CaseNoteService.java

```diff
@@ -381,7 +384,39 @@ public int deleteCaseNotesForOffender(final String offenderIdentifier) {
         repository.deleteOffenderCaseNoteAmendmentsByOffenderIdentifier(offenderIdentifier);
         final var deletedCaseNotesCount = repository.deleteOffenderCaseNoteByOffenderIdentifier(offenderIdentifier);
         log.info("Deleted {} case notes for offender identifier {}", deletedCaseNotesCount, offenderIdentifier);
-        telemetryClient.trackEvent("OffenderDelete", Map.of("offenderNo", offenderIdentifier, "count", String.valueOf(deletedCaseNotesCount)), null);
+        telemetryClient.trackEvent("OffenderDelete", Map.of("offenderNo", offenderIdentifier, "count", valueOf(deletedCaseNotesCount)), null);
         return deletedCaseNotesCount;
     }
+
+    @Transactional
+    @PreAuthorize("hasAnyRole('ROLE_DELETE_CASE_NOTE')")
+    public void softDeleteCaseNote(final String offenderIdentifier, final UUID caseNoteId) {
+        final var caseNote = repository.findById(caseNoteId).orElseThrow(() -> new EntityNotFoundException("Case note not found"));
+        if (!caseNote.getOffenderIdentifier().equalsIgnoreCase(offenderIdentifier)) {
+            throw new ValidationException("case note id not connected with offenderIdentifier");
+        }
+        repository.deleteById(caseNoteId);
+        telemetryClient.trackEvent("CaseNoteSoftDelete",
```

Comment:
think it would be good to include `Secure` in this as this service deals with secure case notes.  Not sure what other events exist, but follow same pattern...